src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,42 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_USER_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkLogin(user);
+        checkAge(user);
+        checkPassword(user);
+        return storageDao.add(user);
+    }
+
+    private void checkPassword(User user) {
+        if (user.getPassword() == null) {
+            throw new RuntimeException("Password can't be null");
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RuntimeException("Password has less than 6 symbols");
+        }
+    }
+
+    private void checkAge(User user) {
+        if (user.getAge() == null) {
+            throw new RuntimeException("Age can't be null");
+        }
+        if (user.getAge() < MIN_USER_AGE) {
+            throw new RuntimeException("Age less then 18");
+        }
+    }
+
+    private void checkLogin(User user) {
+        if (user.getLogin() == null) {
+            throw new RuntimeException("Login can't be null");
+        }
+        if (user.equals(storageDao.get(user.getLogin()))) {
+            throw new RuntimeException("User is already exist");
+        }
```

Comment:
related to previous comment:
test should fail because of this line, we should check that user with such login already exists, not that users are equal

Reply:
> 

Okay, I'm not really understanding what I need to do.